packages/runtime-core/src/renderer.ts

```diff
@@ -813,7 +814,13 @@ function baseCreateRenderer(
     }
     parentComponent && toggleRecurse(parentComponent, true)
 
-    if (__DEV__ && isHmrUpdating) {
+    if (
+      (__DEV__ && isHmrUpdating) ||
+      (dynamicChildren &&
+        prevDynamicChildren &&
+        dynamicChildren.length === 0 &&
```

Comment:
This change is incorrect and may lead to unnecessary updates since empty `dynamicChildren` means no patch is needed in the fast path.
I think we should not track block if the entire slot node is static.